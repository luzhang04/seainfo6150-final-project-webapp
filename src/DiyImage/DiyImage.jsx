
import React from "react";
import PropTypes from "prop-types";
import styles from "./DiyImage.module.css";

const DiyImage = (props) => {
  return <img className={styles.image} alt={props.name} src={props.url} />;
};

DiyImage.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
export default DiyImage;
