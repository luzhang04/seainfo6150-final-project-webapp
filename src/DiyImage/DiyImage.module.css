.image {
	margin: 0 20px 20px 0;
	width: 300px;
	height: 300px;
}
