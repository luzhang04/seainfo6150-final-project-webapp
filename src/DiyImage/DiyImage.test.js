import React from "react";
import { render } from "@testing-library/react";
import DiyImage from "./DiyImage.jsx";

describe("DiyImage tests", () => {
  it("renders correctly", () => {
    const { container } = render(<DiyImage name="the title" url="https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/splatter-painted-easter-egg-designs-1616169765.jpg?crop=0.9514285714285714xw:1xh;center,top&resize=980:*"/>);
    expect(container).toMatchSnapshot();
  });
});