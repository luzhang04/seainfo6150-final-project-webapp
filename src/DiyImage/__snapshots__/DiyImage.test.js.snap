// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiyImage tests renders correctly 1`] = `
<div>
  <img
    alt="the title"
    class="image"
    src="https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/splatter-painted-easter-egg-designs-1616169765.jpg?crop=0.9514285714285714xw:1xh;center,top&resize=980:*"
  />
</div>
`;
